dowhy/causal_model.py

```diff
@@ -1,514 +1,518 @@
-            interpreter(self, **kwargs).interpret()
-
-    def summary(self, print_to_stdout=False):
-        """Print a text summary of the model.
-
-        :returns: a string containining the summary
-
-        """
-        summary_text = "Model to find the causal effect of treatment {0} on outcome {1}".format(self._treatment, self._outcome)
-        self.logger.info(summary_text)
-        if print_to_stdout:
-            print(summary_text)
-        return summary_text
-
-    def refute_graph(self, k= 1, method_name =None, independence_constraints = None ):
-        """
-        Check if the dependencies in input graph matches with the dataset - 
-        ( X ⫫ Y ) | Z 
-        where X and Y are considered as singleton sets currently
-        Z can have multiple variables
-        :param k: number of covariates in set Z 
-        :param method_name: name of method to test conditional independece in data
-        :param independence_constraints: list of implications to be test input by the user in the format 
-        [(x,y,(z1,z2)),
-        (x,y, (z3,))
-        ]
-        : returns: an instance of GraphRefuter class
-        """
-        
-        refuter = GraphRefuter(data = self._data, method_name= method_name)
-
-        if independence_constraints is None:
-            all_nodes=list(self._graph.get_all_nodes(include_unobserved=False))
-            num_nodes = len(all_nodes)
-            array_indices = list(range(0,num_nodes))
-            all_possible_combinations = list(combinations(array_indices, 2))  #Generating sets of indices of size 2 for different x and y
-            conditional_independences=[]
-            self.logger.info("The followed conditional independences are true for the input graph")
-            for combination in all_possible_combinations:  #Iterate over the unique 2-sized sets [x,y]
-                i=combination[0]
-                j=combination[1]
-                a=all_nodes[i]
-                b=all_nodes[j]
-                if(i < j ):
-                    temp_arr = all_nodes[:i] + all_nodes[i+1:j]+ all_nodes[j+1:]
-                else:
-                    temp_arr = all_nodes[:j] + all_nodes[j+1:i]+ all_nodes[i+1:]
-                k_sized_lists=list(combinations(temp_arr,k))
-                for k_list in k_sized_lists:
-                    if self._graph.check_dseparation([str(a)], [str(b)], k_list) == True :
-                        self.logger.info(" %s and %s are CI given %s ", a, b, k_list)
-                        conditional_independences.append([a, b, k_list])
-    
-            res = refuter.refute_model(independence_constraints = conditional_independences)
-
-        else:
-            res = refuter.refute_model(independence_constraints = independence_constraints)
-        
-        self.logger.info(refuter._refutation_passed)
-        
-        return res
-
-
-
-
-
+""" Module containing the main model class for the dowhy package.

```

Comment:
somehow git is thinking that you replaced the entire file. This needs to be corrected. 
Can you:
1. Copy the current version to a temp file outside the repo
2. Revert this file to the version that you checked out from master.
3. Then only change the relevant parts using code from the temp file.